datadog_checks_dev/datadog_checks/dev/tooling/commands/test.py

```diff
@@ -6,92 +6,51 @@
-        )
-        for check in checks
-    )
-
-
-def testable_files(files):
-    """
-    Given a list of files, return the files that have an extension listed in TESTABLE_FILE_EXTENSIONS
-    """
-    return [f for f in files if f.endswith(TESTABLE_FILE_EXTENSIONS)]
-
-
-def get_changed_checks():
-    # Get files that changed compared to `master`
-    changed_files = files_changed()
-
-    # Filter by files that can influence the testing of a check
-    changed_files[:] = testable_files(changed_files)
-
-    return {line.split('/')[0] for line in changed_files}
+def display_envs(check_envs):
+    for check, envs in check_envs:
+        echo_success('`{}`:'.format(check))
+        for e in envs:
+            echo_info('    {}'.format(e))
 
 
 @click.command(
     context_settings=CONTEXT_SETTINGS,
     short_help='Run tests'
 )
 @click.argument('checks', nargs=-1)
+@click.option('--style', '-s', is_flag=True, help='Run only style checks')
 @click.option('--bench', '-b', is_flag=True, help='Run only benchmarks')
 @click.option('--cov', '-c', 'coverage', is_flag=True, help='Measure code coverage')
 @click.option('--cov-missing', '-m', is_flag=True, help='Show line numbers of statements that were not executed')
 @click.option('--pdb', 'enter_pdb', is_flag=True, help='Drop to PDB on first failure, then end test session')
 @click.option('--debug', '-d', is_flag=True, help='Set the log level to debug')
 @click.option('--verbose', '-v', count=True, help='Increase verbosity (can be used additively)')
+@click.option('--list', '-l', 'list_envs', is_flag=True, help='List available test environments')
 @click.option('--changed', is_flag=True, help='Only test changed checks')
 @click.option('--cov-keep', is_flag=True, help='Keep coverage reports')
-def test(checks, bench, coverage, cov_missing, enter_pdb, debug, verbose, changed, cov_keep):
+def test(checks, style, bench, coverage, cov_missing, enter_pdb, debug, verbose, list_envs, changed, cov_keep):
     """Run tests for Agent-based checks.
 
     If no checks are specified, this will only test checks that
     were changed compared to the master branch.
-    """
-    root = get_root()
 
-    if checks:
-        checks_to_test = get_testable_checks() & set(checks)
-        if changed:
-            checks_to_test = checks_to_test & get_changed_checks()
-        # Retain order
-        final_checks = []
-        for check in checks:
-            if check in checks_to_test:
-                final_checks.append(check)
-                checks_to_test.remove(check)
-        checks = final_checks
-    else:
-        checks = sorted(get_testable_checks() & get_changed_checks())
+    You can also select specific comma-separated environments to test like so:
 
-    if not checks:
-        echo_info('No checks to test!')
+    \b
+    $ ddev test mysql:mysql57,maria10130
+    """
+    if list_envs:
+        check_envs = get_tox_envs(checks, every=True, sort=True)
```

Comment:
how is `ddev test --list` listing only changed-checks ?

Reply:
https://github.com/DataDog/integrations-core/pull/2312/files#diff-4cf6877fa2333401bb7c460cbff7f43eR19